xdsl/dialects/llvm.py

```diff
@@ -1735,56 +1736,159 @@ def __init__(self, val: Operation | SSAValue, res_type: Attribute):
         )
 
 
+floatingPointLike = AnyOf([Float16Type, Float32Type, Float64Type])
+
+
+class AbstractFloatArithOp(IRDLOperation, ABC):
+    T: ClassVar = VarConstraint("T", floatingPointLike)
+
+    lhs = operand_def(T)
+    rhs = operand_def(T)
+    res = result_def(T)
+
+    fastmathFlags = prop_def(FastMathAttr, default_value=FastMathAttr(None))
+
+    traits = traits_def(Pure())
+
+    def __init__(
+        self,
+        lhs: SSAValue | Operation,
+        rhs: SSAValue | Operation,
+        fast_math: FastMathAttr | FastMathFlag | None = None,
+        attrs: dict[str, Attribute] | None = None,
+    ):
+        if isinstance(fast_math, FastMathFlag | str | None):
+            fast_math = FastMathAttr(fast_math)
+
+        super().__init__(
+            operands=[lhs, rhs],
+            result_types=[SSAValue.get(lhs).type],
+            properties={"fastmathFlags": fast_math},
+            attributes=attrs if attrs is not None else {},
+        )
+
+    def print(self, printer: Printer):
```

Comment:
I don't think this is necessary, the assembly format from before should still work

Reply:
`xdsl.utils.exceptions.PyRDLOpDefinitionError: ('Error during the parsing of the assembly format: ', (Span[38:38](text=''), "fastmathFlags properties are missing from the declarative format. If this is intentional, consider using 'ParsePropInAttrDict' IRDL option."))`

Reply:
Have you considered using the 'ParsePropInAttrDict' IRDL option?

Reply:
Ah, no it doesnt! Because `fastmath=none` flags are not supposed to be printed at all!

Reply:
Hmmmm, yeah maybe we could hide attributes if they are the default value

Reply:
This should have been happening anyway, but I think I must have missed a case somehow

Reply:
#3783

Reply:
This is now merged